meteor/lib/api/triggers/actionFilterChainCompilers.ts

```diff
@@ -651,32 +651,54 @@ export function compileAdLibFilter(
 			}
 		}
 
-		let result: IWrappedAdLib[] = []
+		const partMap = new Map<PartId, DBPart>()
+		{
+			if (partFilter?.length) {
+				Parts.find({
+					_id: { $in: partFilter },
+				}).forEach((part) => {
+					partMap.set(part._id, part)
+				})
+			}
+		}
+
+		let adlibs: IWrappedAdLib[] = []
 
 		{
-			result = [
+			adlibs = [
 				...rundownBaselineAdLibItems,
 				...rundownBaselineAdLibActions,
 				...adLibPieces,
 				...adLibActions,
 				...clearAdLibs,
 				...stickyAdLibs,
 			]
-			result = result.sort((a, b) => a._rank - b._rank || compareLabels(a.label, b.label))
+
+			// Sort the adliba:
+			adlibs = sortAdlibs(
+				adlibs.map((adlib) => ({
+					adlib: adlib,
+					label: adlib.label,
+					adlibRank: adlib._rank,
+					adlibId: adlib._id,
+					partRank: (adlib.partId && partMap.get(adlib.partId))?._rank ?? null,
+					segmentRank: 0, // TODO: should we take the segment rank into account?
```

Comment:
I'm afraid we do (otherwise the result will be different than the ultimate result in the AdLibPanel.tsx) and then on top of that, we also need to consider `Rundown._rank` here as well - which we didn't have to in the AdLibPanel.tsx, because that already operated on a sorted list of Rundowns and Segments.